### ISSUE TITLE
connect ETIMEDOUT error is not caught in error event listener

### ISSUE THREAD

**Issue by aligatorr89** [DoctorMcKay/node-steam-user#496]
## Describe the bug
Whenever `connect ETIMEDOUT` is thrown out of SteamUser client, it is not caught in error event listener. It is causing server to shut down. I am using `httpProxy` and whenever 1 proxy is faulty it breaks whole server.

```
    this.steamUser = new SteamUser({
      httpProxy,
      promptSteamGuardCode: false,
    });
```

Expected solution: `connect ETIMEDOUT` can be caught on 'error' listener.

```
    this.steamUser.on('error', (error: any) => {
      return this.emit('error', { code: 'LOGIN', error });
    });
```

## Versions
4.24.3

**Comment by aligatorr89** [comment]
@DoctorMcKay here are more details about error that is causing uncaughtException

```
"Error: connect ETIMEDOUT $IP_ADDR
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16)"
```

```
address: "$IP_ADDR"
code: "ETIMEDOUT"
errno: -110
port: PORT
syscall: "connect"
```

**Comment by Sadzurami** [comment]
@aligatorr89 you are using outdated lib version.

It may be fixed in 5.1.0. Try to update.

**Comment by aligatorr89** [comment]
> @aligatorr89 you are using outdated lib version.
> 
> It may be fixed in 5.1.0. Try to update.

@Sadzurami I will try it and close this bug report asap!

**Comment by aligatorr89** [comment]
> connect ETIMEDOUT

Since version update to 5.1.0 there are no more uncaught errors. Tyvm!
